### PR TITLE
Rewrite scaleDown to fix a series of bugs

### PR DESCRIPTION
**Pull request by jeanschmidt**
On scaleDown:

- [FIX] Guaranteed stop the runners from the oldest to the newest, avoiding having runners for too long;
- [FIX] Fixed bug where a runner could be removed from GHA but kept running on AWS;
- [IMPROVED] Try to maintain always a minimum of `minAvailableRunners` runners;

### CODE REVIEW

**Review by seemethere — APPROVED** [review]
LGTM, want to get a second review from @izaitsevfb

File: terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-down.test.ts
Comment (izaitsevfb) [inline terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-down.test.ts:156]: nit: Can take advantage of the types system to ensure that runner ids and names used later in the test come from the set of the runners defined here.

File: terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-down.test.ts
Comment (izaitsevfb) [inline terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-down.test.ts:158]: nit: The unsafe cast and lint suppression could be extracted into a simple function to avoid code duplication, something like this:
```
/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
const mockRunner(runnerDef:any): GhRunner = runnerDef as GhRunner;
```

File: terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-down.test.ts
Comment (izaitsevfb) [inline terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-down.test.ts:184]: nit: 
```
dateRef
          .clone()
          .subtract(minimumRunningTimeInMinutes + x, 'minutes')
          .toDate()
```
can be extracted as a reusable local function.

File: terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-down.test.ts
Comment (izaitsevfb) [inline terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-down.test.ts:419]: Why not do `const { awsR } = getRunnerPair(...)` to avoid lint warning?

**Review by izaitsevfb — APPROVED** [review]
I've added a bunch of comments, they are basically nits, mostly FYI, something to consider in the future. Overall everything looks pretty good, the code is clean and easy to understand.

**Pull request merged**
